mlir/include/mlir/IR/DialectResourceBlobManager.h

```diff
@@ -93,9 +93,14 @@ class DialectResourceBlobManager {
     return HandleT(&entry, dialect);
   }
 
+  /// Provide access to all the registered blobs via a callable. During access
+  /// the blobs are guaranteed to remain unchanged.
```

Comment:
You can guarantee that the map isn't changed, but the blobs themselves aren't protected right?

Reply:
hmm, good point. I guess better wording: "the blob map is guaranteed to remain unchanged".

if one is evil, they could hold a `BlobEntry` for a while and modify it, etc, but this is pretty much out of scope of this (any?) API.

Reply:
rephrased a bit, thanks!